### PR TITLE
Fix conflicts with low U_xxxx key codes

### PR DESCRIPTION
**Pull request by mcdurdin**

### CODE REVIEW

File: web/source/kmwcallback.js [hunk 1]
```diff
@@ -292,7 +292,7 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
 
       if(e.vkCode > 255) keyCode = e.vkCode;           // added to support extended (touch-hold) keys for mnemonic layouts
         
-      if(e.LisVirtualKey || keyCode > 255)            // added keyCode test for same reason
+      if(e.LisVirtualKey)
```
Comment (mcdurdin) [inline web/source/kmwcallback.js]: Fixes the match on key codes > 255 (point 2 in issue #160)

File: web/source/kmwosk.js [hunk 2]
```diff
@@ -700,8 +700,8 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
       // TODO:  Refactor the overloading of the 'n' parameter here into separate methods.
 
       // Test for fall back to U_xxxx key id - For this first test, we refer to Unicode values.
-      if((keyName.substr(0,2) == 'U_') && (n > osk.keyCodes['K_SPACE']) &&  !(n>127 && n<160)) { // 127 - 160 refer to Unicode control codes.
-        ch=String.fromCharCode(n);
+      if((keyName.substr(0,2) == 'U_') && ((keyShiftState == 0) || (n > osk.keyCodes['K_SPACE']) &&  !(n>127 && n<160))) { // 127 - 160 refer to Unicode control codes.
+        ch=String.fromCharCode(parseInt(keyName.substr(2,6),16));
```
Comment (mcdurdin) [inline web/source/kmwosk.js]: Fixes point 1 in #160

File: web/source/kmwosk.js [hunk 3]
```diff
@@ -700,8 +700,8 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
       // TODO:  Refactor the overloading of the 'n' parameter here into separate methods.
```
Comment (jahorton) [inline web/source/kmwosk.js]: Is this related to the changes with 'n' below in any fashion?

File: web/source/kmwosk.js (hunk 2 above)
Comment (jahorton) [inline web/source/kmwosk.js]: How is `keyShiftState` related to this?  Also, I'm having difficulty recognizing the conditions at play here as they related to #160.  Maybe the best question to ask:  where are ` VKDictionary` entries made?

File: web/source/kmwosk.js (hunk 3 above)
Comment (mcdurdin) [inline web/source/kmwosk.js]: Not really; although I guess the misuse of the `n` parameter below in the old code was one of the overloads of the parameter.

File: web/source/kmwosk.js [hunk 4]
```diff
@@ -699,9 +708,15 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
...
 
-      // Test for fall back to U_xxxx key id - For this first test, we refer to Unicode values.
-      if((keyName.substr(0,2) == 'U_') && (n > osk.keyCodes['K_SPACE']) &&  !(n>127 && n<160)) { // 127 - 160 refer to Unicode control codes.
-        ch=String.fromCharCode(n);
+      // Test for fall back to U_xxxx key id - For this first test, we ignore the key code
+      // and use the keyName
+      if((keyName.substr(0,2) == 'U_')) { 
+        ch=String.fromCharCode(parseInt(keyName.substr(2,6),16));
+        if(ch < 32 || (ch > 127 && ch < 160)) {
```
Comment (jahorton) [inline web/source/kmwosk.js]: Shouldn't this read `if(ch <= 32 || (ch > 127 && ch < 160)) {`?  The original test was if `n > osk.keyCodes['K_SPACE']), where `K_SPACE` should probably be 32, the ASCII code for SPACE?

File: web/source/kmwosk.js [hunk 5]
```diff
@@ -792,14 +807,12 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
...
-            // Key code will be Unicode value for U_xxxx keys
-            if(keyName.substr(0,2) == 'U_')
-            {
-              var tUnicode=parseInt(keyName.substr(2),16);
-              if(!isNaN(tUnicode)) Lkc.Lcode=tUnicode;
-            }
+            // Special case for U_xxxx keys. This vk code will never be used
+            // in a keyboard, so we use this to ensure that keystroke processing
+            // occurs for the key.
+            Lkc.Lcode = 1;
```
Comment (jahorton) [inline web/source/kmwosk.js]: Are we saying that this assigned code never gets utilized, but merely ensures we successfully pass any later `if(!Lkc.Lcode)` tests?

Comment (mcdurdin) [inline web/source/kmwosk.js]: Correct

**Pull request merged**